### DISCUSSION TITLE
Nextjs 13: Layout on demand revalidation

### DISCUSSION THREAD

**Discussion by Sveinki10** [vercel/next.js#47445] [Help]
### Summary

I have a project where I have a global layout which has a header and a footer. The data for these components come from a headless CMS.

I am looking for a way to revalidate this layout on demand without having to revalidate all the pages this layout wraps.

Is there perhaps a way to do that using nextjs 13 new experimental appDirectory?

**Comment by RaffoNin** [comment]
Not an answer but I'm facing the same issue. Revalidating all pages is fine for me because data on footer and navbar should rarely be updated but res.revalidate doesn't support revalidating ALL pages. 

I don't want to manually add a page whenever I make a new route. It's also a drag revalidating every slug.

**Comment by stevenprinsloo** [comment]
So you can actually revalidate a layout file using [`revalidatePath`](https://nextjs.org/docs/app/api-reference/functions/revalidatePath). It takes a second parameter `type`.

This can either be page or layout. If you specify layout, it will:

> This will revalidate any URL that matches the provided layout file on the next page visit.

See docs [here](https://nextjs.org/docs/app/api-reference/functions/revalidatePath#revalidating-a-layout-path)
